src/hooks/spoofing/write_player_gamer_data_node.cpp

```diff
@@ -3,36 +3,37 @@
 
 namespace big
 {
-	void hooks::write_player_gamer_data_node(rage::netObject* player, CPlayerGamerDataNode* node)
-	{
-		g_hooking->get_original<write_player_gamer_data_node>()(player, node);
+    void hooks::write_player_gamer_data_node(rage::netObject* player, CPlayerGamerDataNode* node)
+    {
+        g_hooking->get_original<write_player_gamer_data_node>()(player, node);
 
-		if (g.spoofing.spoof_crew_data)
-		{
-			node->m_clan_data.m_clan_member_id = 1;
-			node->m_clan_data.m_clan_id = 41564112;
-			node->m_clan_data.m_clan_id_2 = 41564112;
-			node->m_clan_data.m_clan_color = 420;
-			node->m_clan_data.m_clan_member_count = 15;
-			node->m_clan_data.m_clan_created_time = 420;
+        // move these variables to a separate function
+        const bool spoof_rockstar_id = g.spoofing.spoof_rockstar_id;
+        const bool spoof_rockstar_dev = spoof_rockstar_id && g.spoofing.spoof_rockstar_dev;
+        const bool spoof_rockstar_qa = spoof_rockstar_id && g.spoofing.spoof_rockstar_qa;
 
-			strcpy(node->m_clan_data.m_clan_tag, g.spoofing.crew_tag.c_str());
-			node->m_clan_data.m_is_system_clan = g.spoofing.rockstar_crew;
-			node->m_clan_data.m_is_clan_open = g.spoofing.square_crew_tag;
+        if (g.spoofing.spoof_crew_data)
+        {
+            node->m_clan_data.m_clan_member_id = 1;
+            node->m_clan_data.m_clan_id = 41564112;
+            node->m_clan_data.m_clan_id_2 = 41564112;
+            node->m_clan_data.m_clan_color = 420;
+            node->m_clan_data.m_clan_member_count = 15;
+            node->m_clan_data.m_clan_created_time = 420;
 
-			if (g.spoofing.rockstar_crew)
-			{
-				strcpy(node->m_clan_data.m_clan_name, "Rockstar");
-			}
-		}
+            strcpy(node->m_clan_data.m_clan_tag, g.spoofing.crew_tag.c_str());
+            node->m_clan_data.m_is_system_clan = g.spoofing.rockstar_crew;
+            node->m_clan_data.m_is_clan_open = g.spoofing.square_crew_tag;
 
-		node->m_is_cheater = g.spoofing.spoof_cheater;
+            if (g.spoofing.rockstar_crew)
+            {
+                strcpy(node->m_clan_data.m_clan_name, "Rockstar");
+            }
+        }
 
-		// spoof r* dev and qa only when spoofing rid because every player in the session will send metrics if you join with that enabled
-		if (g.spoofing.spoof_rockstar_id)
-		{
-			node->m_is_rockstar_dev = g.spoofing.spoof_rockstar_dev;
-			node->m_is_rockstar_qa = g.spoofing.spoof_rockstar_qa;
-		}
-	}
+        node->m_is_cheater = g.spoofing.spoof_cheater;
+
+        node->m_is_rockstar_dev = spoof_rockstar_dev;
+        node->m_is_rockstar_qa = spoof_rockstar_qa;
```

Comment:
There was a reason these were wrapped in an if-statement that checked if your Rockstar ID was spoofed.
It's because that setting these will cause players to send out metrics snitching on your RID.

There was even a comment in the old code explaining this.
https://github.com/YimMenu/YimMenu/blob/86c44f38b5a480cb13138d177a64659f4aa3ac71/src/hooks/spoofing/write_player_gamer_data_node.cpp#L31

Reply:
Should be fixed.